<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>NoiseFigureControlDialog</class>
 <widget class="QDialog" name="NoiseFigureControlDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>514</width>
    <height>531</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>Liberation Sans</family>
    <pointsize>9</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>Noise Source Control Settings</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QGroupBox" name="programsBox">
        <property name="title">
         <string>Programs / scripts</string>
        </property>
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="powerOnCommandLabel">
           <property name="text">
            <string>Power on</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="powerOnCommand">
           <property name="toolTip">
            <string>Program or script to execute to turn on the noise source's power</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="powerOffCommand">
           <property name="toolTip">
            <string>Program or script to execute to turn off the noise source's power</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="powerOffCommandLabel">
           <property name="text">
            <string>Power off</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="visaBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>VISA / SCPI</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLabel" name="deviceLabel">
             <property name="text">
              <string>Device</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="device">
             <property name="toolTip">
              <string>VISA device address to send SCPI commands to.
E.g:
USB0::0x1234::0xabcd::DP8C1234::0::INSTR
TCPIP0::192.168.0.10::inst0::INSTR</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLabel" name="powerOnSCPILabel">
           <property name="text">
            <string>Power On SCPI</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPlainTextEdit" name="powerOnSCPI">
           <property name="toolTip">
            <string>SCPI commands to power on noise source.
Lines beginning with # are comments</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="powerOffSCPILabel">
           <property name="text">
            <string>Power Off SCPI</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPlainTextEdit" name="powerOffSCPI">
           <property name="toolTip">
            <string>SCPI commands to power off noise source</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="timingBox">
        <property name="title">
         <string>Timing</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="delayLabel">
           <property name="text">
            <string>Delay in seconds betwen power on/off and measurement</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="delay">
           <property name="toolTip">
            <string>Delay in seconds between the noise source being powered on or off and when the measurement starts</string>
           </property>
           <property name="decimals">
            <number>3</number>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>powerOnCommand</tabstop>
  <tabstop>powerOffCommand</tabstop>
  <tabstop>device</tabstop>
  <tabstop>powerOnSCPI</tabstop>
  <tabstop>powerOffSCPI</tabstop>
  <tabstop>delay</tabstop>
 </tabstops>
 <resources>
  <include location="../../../sdrgui/resources/res.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>NoiseFigureControlDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>NoiseFigureControlDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
